testsuite/tests/typing-modules/inclusion_errors.ml

```diff
@@ -896,7 +897,8 @@ Error: Signature mismatch:
          val f : < m : [< `Foo ] > -> unit
        The type "< m : 'a. [< `Foo ] as 'a > -> unit"
        is not compatible with the type "< m : [< `Foo ] > -> unit"
-       Types for method "m" are incompatible
+       The method "m" has type "'b. [< `Foo ] as 'b",
```

Comment:
The variable names are incoherent with the previous line which is a sign that there was not enough naming context sharing.

Reply:
It took me a while to decipher what you wrote, and I would guess that it will be obscure to @jchavarri as well. I think that you are pointing out that there is a bug in the implementation, related to the handling of the printing context. This results in `'a . ... as 'a` above being printed as `'b . ... as 'b` below, when the naming choice should remain consistent.

Reply:
(If this is correct, I would be interested in your help to fix it, because it is very unclear for someone not familiar with these internal details. I looked and my guess is that there is no obvious fix because the code needs to be massaged a bit to do things correctly.)

Reply:
Thanks for the explanations. I will take a stab at fixing the problem, and come up with hopefully specific questions if I find any blockers.

Reply:
@gasche Sorry for being unclear, I am looking at it; I just wanted to point out that there was some issues with the complicated invariants of printtyp before you merge the PR.